packages/roosterjs-react-common/lib/utils/ImageManager.ts

```diff
@@ -0,0 +1,83 @@
+import { css } from '../utils/ReactUtil';
+import { Base64Svgs } from '../resources/Images';
+import { Editor } from 'roosterjs-editor-core';
+import * as Styles from '../scss/core.scss.g';
+
+export interface ImageManagerInteface {
+    upload: (editor: Editor, image: File) => HTMLElement;
+}
+
+export interface ImageManagerOptions {
+    uploadImage: (dataUrl: string) => Promise<string>;
+    createImagePlaceholder?: (editor: Editor, image: File) => HTMLElement;
+    placeHolderImageClassName?: string;
+}
+
+export default class ImageManager implements ImageManagerInteface {
+    private static Id: number = 0;
+
+    private options: ImageManagerOptions;
+
+    constructor(options: ImageManagerOptions) {
+        this.options = { ...options };
+        this.options.createImagePlaceholder = this.options.createImagePlaceholder || this.defaultCreateImagePlaceholder;
+    }
+
+    public upload(editor: Editor, image: File): HTMLElement {
+        const placeholder = this.options.createImagePlaceholder(editor, image);
+        if (placeholder === null) {
+            return null;
+        }
+
+        // note: add ID for debug/identification (possibly handle undo/redo scenarios later)
+        let preserveId = true;
+        if (!placeholder.id) {
+            preserveId = false;
+            placeholder.id = `paste-image-placeholder-${ImageManager.Id++}`;
+        }
+
+        let reader = new FileReader();
```

Comment:
Looks like in VSTS we use a different rest endpoint (attachments) for uploading images using the toolbar, and for pasted images we use uploadAttachmentBinary (MVC) endpoint.
The attachments rest endpoint actually just needs the file object and not the dataURL:
![image](https://user-images.githubusercontent.com/4613474/39602092-fc513f52-4ed8-11e8-9300-bc6ba7d7e0b5.png)

I can write a different implementation of ImageManager, but in that case we would be nice to expose the spinner SVG.